DATA SCIENCE- ASSIGNMENT ONE

NOTE: Some answers of the questions are being done on the MS-Word file itself.

Q7) Calculate Mean, Median, Mode, Variance, Standard Deviation, Range &     comment about the values / draw inferences, for the given dataset
-	For Points,Score,Weigh>
Find Mean, Median, Mode, Variance, Standard Deviation, and Range and also Comment about the values/ Draw some inferences.
Use Q7.csv file 


In [2]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as pyplot
import seaborn as sns

In [3]:
dataQ7 = pd.read_csv(r"Q7.csv")

In [9]:
dataQ7

Cars  Points  Score  Weigh
0             Mazda RX4    3.90  2.620  16.46
1         Mazda RX4 Wag    3.90  2.875  17.02
2            Datsun 710    3.85  2.320  18.61
3        Hornet 4 Drive    3.08  3.215  19.44
4     Hornet Sportabout    3.15  3.440  17.02
5               Valiant    2.76  3.460  20.22
6            Duster 360    3.21  3.570  15.84
7             Merc 240D    3.69  3.190  20.00
8              Merc 230    3.92  3.150  22.90
9              Merc 280    3.92  3.440  18.30
10            Merc 280C    3.92  3.440  18.90
11           Merc 450SE    3.07  4.070  17.40
12           Merc 450SL    3.07  3.730  17.60
13          Merc 450SLC    3.07  3.780  18.00
14   Cadillac Fleetwood    2.93  5.250  17.98
15  Lincoln Continental    3.00  5.424  17.82
16    Chrysler Imperial    3.23  5.345  17.42
17             Fiat 128    4.08  2.200  19.47
18          Honda Civic    4.93  1.615  18.52
19       Toyota Corolla    4.22  1.835  19.90
20        Toyota Corona    3.70  2.465  20.01
21     Dodge Challenger    2.76  3.520  16.87
22          AMC Javelin    3.15  3.435  17.30
23           Camaro Z28    3.73  3.840  15.41
24     Pontiac Firebird    3.08  3.845  17.05
25            Fiat X1-9    4.08  1.935  18.90
26        Porsche 914-2    4.43  2.140  16.70
27         Lotus Europa    3.77  1.513  16.90
28       Ford Pantera L    4.22  3.170  14.50
29         Ferrari Dino    3.62  2.770  15.50
30        Maserati Bora    3.54  3.570  14.60
31           Volvo 142E    4.11  2.780  18.60

In [10]:
dataQ7.head()

Cars  Points  Score  Weigh
0          Mazda RX4    3.90  2.620  16.46
1      Mazda RX4 Wag    3.90  2.875  17.02
2         Datsun 710    3.85  2.320  18.61
3     Hornet 4 Drive    3.08  3.215  19.44
4  Hornet Sportabout    3.15  3.440  17.02

In [11]:
dataQ7.shape

(32, 4)

In [4]:
dataQ7.describe()

Points      Score      Weigh
count  32.000000  32.000000  32.000000
mean    3.596563   3.217250  17.848750
std     0.534679   0.978457   1.786943
min     2.760000   1.513000  14.500000
25%     3.080000   2.581250  16.892500
50%     3.695000   3.325000  17.710000
75%     3.920000   3.610000  18.900000
max     4.930000   5.424000  22.900000

In [ ]:
dataQ7.median()

Points     3.695
Score      3.325
Weigh     17.710
dtype: float64

In [ ]:
dataQ7.mode()

Points  Score  Weigh
0    3.07   3.44  17.02
1    3.92    NaN  18.90

In [ ]:
print(dataQ7.max())

Points     4.930
Score      5.424
Weigh     22.900
dtype: float64


In [ ]:
print(dataQ7.min())

Points     2.760
Score      1.513
Weigh     14.500
dtype: float64


In [ ]:
Q7range = dataQ7.max() - dataQ7.min()

In [ ]:
print(Q7range)

Points    2.170
Score     3.911
Weigh     8.400
dtype: float64


In [ ]:
Q7stds = dataQ7.std()

In [ ]:
print(Q7stds)

Points    0.534679
Score     0.978457
Weigh     1.786943
dtype: float64


In [ ]:
Q7var = dataQ7.var()

In [ ]:
print(Q7var)

Points    0.285881
Score     0.957379
Weigh     3.193166
dtype: float64


In [ ]:
highest_points_cars = dataQ7.loc[dataQ7["Points"].idxmax(), "Cars"]

In [ ]:
print(highest_points_cars)

Honda Civic


In [ ]:
lowest_points_cars = dataQ7.loc[dataQ7["Points"].idxmin(), "Cars"]

In [ ]:
print(highest_points_cars)
print(lowest_points_cars)

Honda Civic
Valiant


In [ ]:
highest_score_cars = dataQ7.loc[dataQ7["Score"].idxmax(), "Cars"]
lowest_score_cars = dataQ7.loc[dataQ7["Score"].idxmin(), "Cars"]
print(highest_score_cars)
print(lowest_score_cars)

Lincoln Continental
Lotus Europa


In [ ]:

highest_weight_cars = dataQ7.loc[dataQ7["Weigh"].idxmax(), "Cars"]
lowest_weight_cars = dataQ7.loc[dataQ7["Weigh"].idxmin(), "Cars"]
print(highest_weight_cars)
print(lowest_weight_cars)



Merc 230
Ford Pantera L


Q 7. INFERENCES:

1. Mean: Mean represents the overall value of a particular list of data. It gives an overall idea abouta set of data being collected.
2. Mode: It represents which of value have seen more number of attendance or which value has been repeated the most from the overall list of elements.
3. Median: It represents the middle-most point of data.
4. Range: Range is the highest and lowest point of any given data.
5. Standard Deviation:
6. Variance: 
 
        The above dataset which has details about cars. There are three details about the car being told is points, score and weight. The Top three cars in the above being Mazda RX4, 
 Mazda RX4 Wag and Datsun 710. Some analysis  is given as below:

        A. Points:

1. Range                : The range of the score is 2.170. The Highest point being 4.930 and the lowest point in the data is 2.760.
2. Mean                 : The mean of the point is 3.5965
3. Median               : Median of the points is 3.695
4. Mode                 : Mode of the points is 3.07.
5. Standard deviation   : The standard deviation of the Points is 0.5347
5. Variance             : The Variance of the Points is 0.2859.

        B. Score:

1. Range                : The range of the score is 3.911. The Highest score being 5.424 and the lowest score in the data is 1.513.
2. Mean                 : The mean of the score is 3.21725
3. Median               : Median of the score is 3.325
4. Mode                 : Mode of the score is 3.44.
5. Standard deviation   : The standarad deviation of the score is 0.9784.
5. Variance             : The Variance of the score is 0.9574.

        C. Weigh:

1. Range                : The range of the score is 8.4. The Highest score being 22.9 and the lowest score in the data is 14.5.
2. Mean                 : The mean of the score is 17.85.
3. Median               : Median of the score is 17.71
4. Mode                 : Mode of the score is 17.02.
5. Standard deviation   : The standarad deviation of the score is 1.79.
5. Variance             : The Variance of the score is 3.19.

        When it comes to points, the minimum value of the points was of Honda Civic and highest value in points was of Valiant.
        The highest score was of Lincoln Continental and lowest score was of Lotus Europa.
        The car with highest weight was Merc 230 and with lowest weight was Ford Pantera L.



Q8) Calculate Expected Value for the problem below

a)	The weights (X) of patients at a clinic (in pounds), are
108, 110, 123, 134, 135, 145, 167, 187, 199
Assume one of the patients is chosen at random. What is the Expected Value of the Weight of that patient?


In [ ]:
x_weights_pounds = (108, 110, 123, 134, 135, 145, 167, 187, 199)
mean_weight_x = np.mean(x_weights_pounds)
print("Mean Weight:", mean_weight_x)

Mean Weight: 145.33333333333334


Q9) Calculate Skewness, Kurtosis & draw inferences on the following data.
      Cars speed and distance 
Use Q9_a.csv

SP and Weight(WT)
Use Q9_b.csv


In [3]:

Q9a_speed = pd.read_csv(r"Q9_a.csv")
Q9b_sp = pd.read_csv(r"Q9_b.csv")

Q9a- CALCULATION OF SKEWNESS AND KURTOSIS OF SPEED AND DISTANCE

In [4]:
speed = Q9a_speed["speed"]
distance = Q9a_speed["dist"]

In [5]:
speed_skewness = speed.skew()
distance_skewness = distance.skew()
speed_kurtosis = speed.kurt()
distance_kurtosis = distance.kurt()

In [6]:
# RESULT of Speed

print("Speed:")
print(f"Skewness: {speed_skewness: .3f}")
print(f"Kurtosis: {speed_kurtosis: .3f}")

Speed:
Skewness: -0.118
Kurtosis: -0.509


i. SPEED- Skewness

The weak negative skewness (-0.118) indicating that the speed data is slightly left-skewed. This means there are more values towards the lower end of the speed range compared to the higher end. However, the calculated skewness is very close to 0, indicating that the distribution is close to middle of skewed graph.

ii. SPEED- Kurtosis

The negative kurtosis (-0.509) means the data might have fewer extreme values. There is not much variations in the speed compared to a normal distribution. It suggests the data leaned in the center and has fewer outliers in the tails.

In [ ]:

# Result of distance

print("\nDistance:")
print(f"Skewness: {distance_skewness: .3f}")
print(f"Kurtosis: {distance_kurtosis: .3f}")


Distance:
Skewness:  0.807
Kurtosis:  0.405



i. DISTANCE- Skewness

The positive skewness of 0.807 indicates that the distance data is highly right-skewed. This means there are more values towards the higher end of the distance range compared to the lower end. It means that shorter distances might be less frequent than longer distances.

ii. DISTANCE- Kurtosis

The positive kurtosis, 0.405, shows that the data has heavier tails than a normal distribution. This means the data might have more extreme values compared to a normal distribution. However, the value is relatively close to 3, means it's not significantly heavier-tailed..

Q 9b. Calculate Skewness, Kurtosis & draw inferences on the following : SP and Weight(WT)

In [7]:
Q9b_sp.columns

Index(['Unnamed: 0', 'SP', 'WT'], dtype='object')

In [8]:
SP = Q9b_sp["SP"]
WT = Q9b_sp["WT"]

In [9]:
SP_skewness = SP.skew()
WT_skewness = WT.skew()
SP_kurtosis = SP.kurt()
WT_kurtosis = WT.kurt()

In [10]:
print("SP:")
print(f"Skewness: {SP_skewness: .3f}")
print(f"Kurtosis: {SP_kurtosis: .3f}")

SP:
Skewness:  1.611
Kurtosis:  2.977


### SP
--Skewness
A positive skewness value of 1.611 indicates that the distribution is skewed to the right. The majority of the data points are concentrated on the left side, with lesser values pulling the mean to the right. There are more data-points concentrated on the left side of the distribution with a few larger values on the right side.

---Kurtosis

A kurtosis value of 2.977 suggests that the distribution is leptokurtic, meaning it has heavier tails and a sharper peak compared to a normal distribution which has a kurtosis of 3. It indicates that the data has more extreme values than a normal distribution.


In [11]:
print("WT:")
print(f"Skewness: {WT_skewness: .3f}")
print(f"Kurtosis: {WT_kurtosis: .3f}")

WT:
Skewness: -0.615
Kurtosis:  0.950


### WT

---Skewness
A negative skew of 0.615 indicates that the distribution is skewed to the left. The majority of the data points are concentrated on the right side, with lesser values pulling the mean to the left. There are more data-points concentrated on the right side of the distribution with a few larger values on the left side.

---Kurtosis

A kurtosis value of 0.95 suggests that the distribution is mesokurtic, meaning it has a peak and tails similar to that of a normal distribution. It indicates that it has a peak and tails similar to that of a normal distribution.

Q. 11.  Suppose we want to estimate the average weight of an adult male in    Mexico. We draw a random sample of 2,000 men from a population of 3,000,000 men and weigh them. We find that the average person in our sample weighs 200 pounds, and the standard deviation of the sample is 30 pounds. Calculate 94%,98%,96% confidence interval?

In [ ]:
sample_mean = 200
sample_std = 30
sample_size = 2000


In [157]:
from scipy.stats import norm

In [ ]:

# Z-Score at 94%
z_score_94pc = norm.ppf((1 + 0.94)/2)   


# Z-Score at 96%
z_score_96pc = norm.ppf((1 + 0.96)/2)   

# Z-Score at 98%
z_score_98pc = norm.ppf((1 + 0.98)/2)   



In [ ]:
# MARGIN OF ERROR

margin_of_error_94pc = z_score_94pc * (sample_std / np.sqrt(sample_size))
margin_of_error_96pc = z_score_96pc * (sample_size / np.sqrt(sample_size))
margin_of_error_98pc = z_score_98pc * (sample_size / np.sqrt(sample_size))

#Confidence_interval
confidence_interval_94pc = (sample_mean - margin_of_error_94pc, sample_mean + margin_of_error_94pc)
confidence_interval_96pc = (sample_mean - margin_of_error_96pc, sample_mean + margin_of_error_96pc)
confidence_interval_98pc = (sample_mean - margin_of_error_98pc, sample_mean + margin_of_error_98pc)

#Results
print("At 94%, confidence interval is", confidence_interval_94pc)
print("At 96%, confidence interval is", confidence_interval_96pc)
print("At 98%, confidence interval is", confidence_interval_98pc)

At 94%, confidence interval is (198.738325292158, 201.261674707842)
At 96%, confidence interval is (108.15355654222209, 291.8464434577779)
At 98%, confidence interval is (95.96256028665124, 304.03743971334876)


Q12)  Below are the scores obtained by a student in tests 
34,36,36,38,38,39,39,40,40,41,41,41,41,42,42,45,49,56
1)	Find mean, median, variance, standard deviation.
2)	What can we say about the student marks? 


In [ ]:
test_score = np.array([34,36,36,38,38,39,39,40,40,41,41,41,41,42,42,45,49,56])

In [ ]:
mean = np.mean(test_score)
median = np.median(test_score)
variance = np.var(test_score)
std_deviation = np.std(test_score)

In [ ]:
print("Mean:", mean)
print("Median:", median)
print("Variance:", variance)

print("Standard deviation:", std_deviation)

Mean: 41.0
Median: 40.5
Variance: 24.11111111111111
Standard deviation: 4.910306620885412


Note: It has been assumed that total marks is 60 and marks given are out of 60

Observations: Mean and Median are close, So there is symmetrical distribution of scores. Standard deviation figure is very less, it means much of the scores are near to mean. The performance of students is showing increasing trend.

Q 20) Calculate probability from the given dataset for the below cases

Data _set: Cars.csv
Calculate the probability of MPG  of Cars for the below cases.
       MPG <- Cars$MPG
a.	P(MPG>38)  
b.	P(MPG<40)  
c.   P(20<MPG<50)


In [ ]:
cars_data = pd.read_csv("Cars.csv")

In [ ]:
mpg = cars_data["MPG"] 

In [ ]:
prob_MOREthan_38 = np.sum(mpg>38)/len(mpg)
prob_LESSthan_40 = np.sum(mpg<40)/len(mpg)
prob_between_20_to_50 = np.sum((mpg>=20) & (mpg <= 50))/len(mpg)

In [ ]:
# PROBABILITIES
print("a. P(MPG>38): ", prob_MOREthan_38)
print("b. P(MPG<40): ", prob_MOREthan_38)
print("c. P(20<MPG<50): ", prob_between_20_to_50)

a. P(MPG>38):  0.4074074074074074
b. P(MPG<40):  0.4074074074074074
c. P(20<MPG<50):  0.8518518518518519


Q 21. Check whether the data follows normal distribution
a.	Check whether the MPG of Cars follows Normal Distribution 
        Dataset: Cars.csv

In [150]:
! pip install scipy
from scipy.stats import t


[notice] A new release of pip is available: 23.3.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [ ]:
# Checking Normal Distribution of MPG with Kolmogorov-Smirnov test
# Cars.csv is already loaded in the variable cars_data

# Shapiro-Wilk test
shapiro_test = stats.shapiro(mpg)
print("Shapiro-Wilk test statistic:", shapiro_test[0])
print("Shapiro-Wilk p-value:", shapiro_test[1])

Shapiro-Wilk test statistic: 0.9779686331748962
Shapiro-Wilk p-value: 0.17639249563217163


The above data is not normally distributed

b.	Check Whether the Adipose Tissue (AT) and Waist Circumference(Waist)  from wc-at data set  follows Normal Distribution 
       Dataset: wc-at.csv

In [ ]:
wc_at = pd.read_csv("wc-at.csv") 

In [ ]:
wc = wc_at["Waist"]
at = wc_at["AT"]

In [ ]:
shapiro_test = stats.shapiro(wc)
print("Shapiro-Wilk test statistic:", shapiro_test[0])
print("Shapiro-Wilk p-value:", shapiro_test[1])

Shapiro-Wilk test statistic: 0.9558579921722412
Shapiro-Wilk p-value: 0.0011704873759299517


In [ ]:
shapiro_test = stats.shapiro(at)
print("Shapiro-Wilk test statistic:", shapiro_test[0])
print("Shapiro-Wilk p-value:", shapiro_test[1])

Shapiro-Wilk test statistic: 0.9523370862007141
Shapiro-Wilk p-value: 0.0006539996829815209


Q 22) Calculate the Z scores of  90% confidence interval,94% confidence interval, 60% confidence interval 

In [ ]:
# to solve the problem it is being assumed that:
population = 1000
population_mean = 100
population_std = 15 # std deviation
confidence_levels = [0.9, 0.94, 0.6]
 
# Z-SCORE-inverse cumulative distribution function
z_score = np.sqrt(level) * np.exp(stats.norm.ppf(1 - (1 - level) / 2))

print(f"Z score: {z_score:.4f}")


Z score: 1.7972


In [ ]:
import pandas as pd
import numpy as np

# Define population mean and standard deviation (replace with actual values if known)
population_mean = 100  # Example mean
population_std = 20  # Example standard deviation

# Set confidence levels
confidence_levels = [0.9, 0.94, 0.6]

for level in confidence_levels:
    # Calculate standard error of the mean
    standard_error = population_std / np.sqrt(100)

    # Calculate Z score using inverse cumulative distribution function (CDF)
    z_score = np.sqrt(level) * np.exp(stats.norm.ppf(1 - (1 - level) / 2))

    

    # Print results
    print(f"Confidence level: {100 * level:.2f}%")
    print(f"Z score: {z_score:.4f}")
   
    print("\n")

Confidence level: 90.00%
Z score: 4.9144


Confidence level: 94.00%
Z score: 6.3589


Confidence level: 60.00%
Z score: 1.7972




Q 23. Calculate the t scores of 95% confidence interval, 96% confidence interval, 99% confidence interval for sample size of 25

In [ ]:
n = 25
confidenct_interval = [0.95, 0.96, 0.99]

# degree of freedom
df = n-1 

#  T-scores in T-distribution table 
from scipy import stats
t_scores = np.array([
    stats.t.ppf(1 - (1 - level) / 2, df)
    for level in confidence_levels
])

# Print results
print("Confidence Level | T-Scores")
print("-----------------|--------")
for level, score in zip(confidence_levels, t_scores):
    print(f"{100 * level:.2f}%           |    {score:.4f}")

Confidence Level | T-Scores
-----------------|--------
90.00%           |    1.7109
94.00%           |    1.9740
60.00%           |    0.8569


  Q 24)   A Government  company claims that an average light bulb lasts 270 days. A researcher randomly selects 18 bulbs for testing. The sampled bulbs last an average of 260 days, with a standard deviation of 90 days. If the CEO's claim were true, what is the probability that 18 randomly selected bulbs would have an average life of no more than 260 days
Hint:  
   rcode   pt(tscore,df)  
 df  degrees of freedom


In [13]:
import numpy as np
from scipy.stats import t

# Given data
sample_mean_lightbulb = 260  # sample mean
population_mean_lightbulb = 270  # population mean
sample_std_lightbulb = 90  # standard deviation of the sample
sample_size_lightbulb = 18  # sample size

# Step 1: Calculate the t-score
t_score = (sample_mean - population_mean) / (sample_std / np.sqrt(sample_size))

# Step 2: Determine degrees of freedom
df_lightbulb = sample_size - 1

# Step 3: Calculate the probability using the cumulative distribution function (CDF)
prob_lightbulb = t.cdf(t_score, df)

print("Conclusion: Hence, probability of 18 randomly selected lightbulbs to have average life of upto 260 days:", prob_lightbulb)


Conclusion: Hence, probability of 18 randomly selected lightbulbs to have average life of upto 260 days: 0.32167253567098364
